### PR TITLE
[spirv] Turn SpirvEvalInfo into a class

### PR DESCRIPTION
**Pull request by antiagainst**
Cleaned up methods on SpirvEvalInfo. Also added a new field or
lvalue/rvalue, which will be used later.

### CODE REVIEW

File: tools/clang/lib/SPIRV/SPIRVEmitter.cpp [hunk 1]
```diff
@@ -3177,13 +3183,14 @@ void SPIRVEmitter::storeValue(const SpirvEvalInfo &lhsPtr,
...
+      const auto subRhsValType = typeTranslator.translateType(
+          field->getType(), rhsVal.getLayoutRule());
       const auto subRhsVal =
           theBuilder.createCompositeExtract(subRhsValType, rhsVal, {index});
       const auto subLhsPtrType = theBuilder.getPointerType(
-          typeTranslator.translateType(field->getType(), lhsPtr.layoutRule),
-          lhsPtr.storageClass);
+          typeTranslator.translateType(field->getType(),
+                                       lhsPtr.getLayoutRule()),
+          lhsPtr.getStorageClass());
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/SPIRVEmitter.cpp]: bad formatting here.

Comment (antiagainst) [reply tools/clang/lib/SPIRV/SPIRVEmitter.cpp]: Also will fix in the next commit

Comment (antiagainst) [reply tools/clang/lib/SPIRV/SPIRVEmitter.cpp]: Well, actually it is not. ClangFormat is happy with this.

Comment (ehsannas) [reply tools/clang/lib/SPIRV/SPIRVEmitter.cpp]: lol, it seems like an odd one! hah
Ok thanks

**Pull request merged**